tiledb/api/src/query/read/callback.rs

```diff
@@ -269,6 +298,40 @@ mod impls {
             Some(vec![])
         }
     }
+
+    impl ReadCallback for (Vec<String>, Vec<u8>) {
+        type Unit = u8;
+        type Intermediate = ();
+        type Final = Self;
+        type Error = std::convert::Infallible;
+
+        fn intermediate_result(
+            &mut self,
+            arg: RawReadOutput<Self::Unit>,
+        ) -> Result<Self::Intermediate, Self::Error> {
+            // Copy validity before VarDataIter consumes arg.
+            self.1.extend_from_slice(
+                &arg.input.validity.as_ref().unwrap().as_ref()[0..arg.nvalues],
+            );
+
+            /* TODO: this `unwrap` is not a sure thing since it depends on user input */
```

Comment:
Similar to the validity buffers, is this actually true?  Can you run the query at all if you don't supply offset buffers for a var-sized attribute?

Reply:
I think the answer is: unwrap() will never fail as core is currently implemented since it has specific checks that these buffers exist or do not exist depending on the field's schema information.

But do we want to guard against core changing those constraints or just comment that we're lying on them?